client/src/pages/CreateWorkout/CreateWorkout.jsx

```diff
@@ -110,9 +120,9 @@ const CreateWorkout = () => {
               </select>
             </div>
 
-            {exercises.length !== 0 && (
+            {workout.exercises?.length !== 0 && (
```

Comment:
Hmm why the `?` - if exercises could potentially be undefined then that would also !== 0, so you'd have to check that it is defined and length !== 0. Alternatively, you could check just do a truthy check: `workout.exercises?.length && (...` b/c both 0 and undefined would be considered falsey and fail that check.